src/v/pandaproxy/schema_registry/seq_writer.cc

```diff
@@ -71,110 +71,299 @@ ss::future<> seq_writer::read_sync_inner() {
+    assert(batch.record_count() > 0);
 
-        kafka::partition_produce_response res
-          = co_await _client.local().produce_record_batch(
-            model::schema_registry_internal_tp, std::move(batch));
+    kafka::partition_produce_response res
+      = co_await _client.local().produce_record_batch(
+        model::schema_registry_internal_tp, std::move(batch));
+    if (res.error_code != kafka::error_code::none) {
+        vlog(
+          plog.error,
+          "Error writing to schema topic: {} {}",
+          res.error_code,
+          res.error_message);
+        throw kafka::exception(res.error_code, *res.error_message);
+    }
 
-        // TODO(Ben): Check the error reporting here
-        if (res.error_code != kafka::error_code::none) {
-            throw kafka::exception(res.error_code, *res.error_message);
+    // Replay the persisted deletions into our store
+    auto applier = consume_to_store(_store);
+    auto offset = res.base_offset;
+    for (auto k : keys) {
+        if (auto skey = std::get_if<schema_key>(&k)) {
+            co_await applier.apply(offset, *skey, std::nullopt);
+        } else if (auto dkey = std::get_if<delete_subject_key>(&k)) {
+            co_await applier.apply(offset, *dkey, std::nullopt);
         }
+        co_await _store.replay(offset);
+        offset++;
+    }
+
+    // TODO: deleting config_key entries, need to remember their sequence
+    // numbers too.  Actually... do config_key entries really need
+    // to be strictly ordered?  We're not allocating anything.
+
+    co_return std::vector<schema_version>();
+}
+
+/// Helper for write methods that need to check + retry if their
+/// write landed where they expected it to.
+///
+/// \param write_at Offset at which caller expects their write to land
+/// \param batch Message to write
+/// \return true if the write landed at `write_at`, else false
+ss::future<bool> seq_writer::produce_and_check(
+  model::offset write_at, model::record_batch batch) {
+    kafka::partition_produce_response res
+      = co_await _client.local().produce_record_batch(
+        model::schema_registry_internal_tp, std::move(batch));
+
+    // TODO(Ben): Check the error reporting here
+    if (res.error_code != kafka::error_code::none) {
+        throw kafka::exception(res.error_code, *res.error_message);
+    }
+
+    auto wrote_at = res.base_offset;
+    if (wrote_at == write_at) {
+        vlog(plog.debug, "seq_writer: Successful write at {}", wrote_at);
+
+        co_return true;
+    } else {
+        vlog(
+          plog.debug,
+          "seq_writer: Failed write at {} (wrote at {})",
+          write_at,
+          wrote_at);
+        co_return false;
+    }
+};
+
+/// Wrapper around client writes, providing sequence number
+/// logic to de-conflict concurrent writes from remote nodes.
+ss::future<schema_id> seq_writer::write_subject_version(
+  subject sub, schema_definition def, schema_type type) {
+    auto do_write =
```

Comment:
This code looks familiar. Is it possible to change the commit sequence to reduce churn?